bin/transform

```diff
@@ -95,11 +97,47 @@ def parse_geographic_columns(gisaid_data: pd.DataFrame) -> pd.DataFrame:
     Expands the string found in the column named `location` in the given
     *df*, creating four new columns. Returns the modified ``pd.DataFrame``.
     """
+    def titlecase_non_whitelisted_tokens(tokens: Union[List[str],
+        pd._libs.missing.NAType]) -> Optional[str]:
+        """
+        Returns a title cased location name from the given location name
+        *tokens*. Ensures that no tokens contained in the *whitelist_tokens* or
+        matching the *whitelist_regex* pattern(s) are converted to title case.
+        """
+        if not isinstance(tokens, list):
+            return
+
+        return ' '.join([
+            w.title()
+            if w not in whitelist_tokens
+            and not re.match(whitelist_regex, w)
+            else w
+            for w in tokens
+        ])
+
     geographic_data = gisaid_data['location'].str.split(r'\s*/\s*', expand=True)
     location_columns = ['region', 'country', 'division', 'location']
 
+    # Manually curate list of tokens and patterns that should not be cast to
+    # title case
+    articles = [
+        'de', 'del', 'di', 'do', 'la', 'le', 'las', 'los', 'nad', 'of', 'op',
+        'the', 'y'
+    ]
+    locations = ['USA', 'DC']
+    whitelist_tokens = articles + locations
+    # Some Francophone locations are hyphen-delimited (e.g. Braine-le-Comte).
+    # These names don't get split into separate tokens, but all pieces are
+    # capitalized by `.title()`. Assume these names are already properly cased,
+    # and prevent them from being transformed.
+    whitelist_regex = r'\S*-le?-?'
```

Comment:
You can avoid this issue and other similar issues by splitting on Unicode word boundaries (`\b`) instead of whitespace. The standard `re` library doesn't support splitting on zero-length matches, but the `regex` library does. We use it elsewhere for better Unicode and regex support. For example:

```python
# Run with: python3 -m doctest -v titlecase.py
import regex

def titlecase(text, articles = {}, abbrev = {}):
    """
    >>> articles = {'a', 'and', 'of', 'the', 'le'}
    >>> abbrev = {'USA', 'DC'}

    >>> titlecase("the night OF THE LIVING DEAD", articles)
    'The Night of the Living Dead'

    >>> titlecase("BRAINE-LE-COMTE, FRANCE", articles)
    'Braine-le-Comte, France'

    >>> titlecase("auvergne-RHÔNE-alpes", articles)
    'Auvergne-Rhône-Alpes'

    >>> titlecase("washington DC, usa", articles, abbrev)
    'Washington DC, USA'
    """
    words = enumerate(regex.split(r'\b', text, flags = regex.V1))

    def changecase(index, word):
        casefold = word.casefold()
        upper = word.upper()
        title = word.title()

        if upper in abbrev:
            return upper
        elif casefold in articles and index != 1:
            return casefold
        else:
            return title

    return ''.join(changecase(i, w) for i, w in words)
```

Reply:
Thanks for writing this handy function. I'll incorporate it into the latest commit. 
Do we want to use `.casefold()` if it might risk losing special characters? e.g. "ß" -> "ss"

Reply:
Ah, good catch! I think the condition should be `casefold in articles` but that branch should `return lower`.

Reply:
I incorporated these changes into a new commit.